File: tasks/3668-restore-finishing-order.ts
// Accepted 921 / 921 testcases passed Sergey Pomortsev submitted at Oct 25, 2025 21:05
// Solution Runtime 0 ms Beats 100.00% Analyze Complexity Memory 59.08 MB Beats 37.22%

const recoverOrder = (order: number[], friends: number[]): number[] => {
  const friendsSet = new Set(friends)
  return order.filter(el => friendsSet.has(el))
}

// or same solution in one line

const recoverOrder2 = (order: number[], friends: number[], friendsSet = new Set(friends)): number[] => order.filter(el => friendsSet.has(el))

File: tasks/3668-restore-finishing-order.go
// Accepted 921 / 921 testcases passed Sergey Pomortsev submitted at Jul 03, 2026 18:44
// Runtime 0 ms Beats 100.00% Memory 5.20 MB Beats 98.02%

func recoverOrder(order []int, friends []int) []int {
  var uniq = make(map[int]struct{})
  var result = []int{}

  for _, val := range friends {
    uniq[val] = struct{}{}
  }

  for _, val := range order {
    if _, ok := uniq[val]; ok {
      result = append(result, val)
    }
  }

  return result
}